src/main/java/com/segment/analytics/android/integrations/optimizelyx/OptimizelyXIntegration.java

```diff
@@ -1,63 +1,199 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.segment.analytics.internal.Utils.isNullOrEmpty;
+import static java.util.concurrent.TimeUnit.SECONDS;
 
 /**
- * Kahuna helps mobile marketers send push notifications and in-app messages.
+ * Optimizely X helps marketers and growth hackers make smart decisions through A/B testing.
  *
- * @see <a href="https://www.optimizely.com/">Pptimizely</a>
- * @see <a href="https://segment.com/docs/integrations/optimizely/">Optimizely Integration</a>
- * @see <a href="http://developers.optimizely.com/android/">Optimizely Android SDK</a>
+ * @see <a href="https://www.optimizely.com/">Optimizely</a>
+ * @see <a href="https://segment.com/docs/destinations/optimizelyx/">Optimizely Integration</a>
+ * @see <a
+ *     href="https://developers.optimizely.com/x/solutions/sdks/introduction/index.html?language=android&platform=mobile">Optimizely
+ *     Android SDK</a>
  */
 public class OptimizelyXIntegration extends Integration<Void> {
-  private static final String OPTIMIZELY_KEY = "Optimizely";
 
-  private final Context context;
+  private static final String OPTIMIZELYX_KEY = "Optimizely X";
+  final NotificationListener listener;
+  private OptimizelyClient client;
   private final Logger logger;
-  private final Analytics analytics;
-
-  // Optimizely must be initialized immediately on launch. So we require the token when creating the
-  // factory, and initialize it in this method.
-//  public static Factory createFactory(final Application application, String token) {
-////    Optimizely.startOptimizelyWithAPIToken(token, application);
-////    return new Factory() {
-////      @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-////        boolean listen = settings.getBoolean("listen", false);
-////        Logger logger = analytics.logger(OPTIMIZELY_KEY);
-////
-////        OptimizelyIntegration integration =
-////            new OptimizelyIntegration(analytics, application, logger);
-////        if (listen) {
-////          Optimizely.addOptimizelyEventListener(integration);
-////        }
-////        return integration;
-////      }
-//
-////      @Override public String key() {
-////      return OPTIMIZELY_KEY;
-//////      }
-//    };
-//  }
-
-  public OptimizelyXIntegration(Analytics analytics, Context context, Logger logger) {
-    this.analytics = analytics;
-    this.context = context;
+  boolean isClientValid = false;
+  boolean trackKnownUsers;
+  static final Options options = new Options().setIntegration(OPTIMIZELYX_KEY, false);
+  private Map<String, String> attributes = new HashMap<>();
+  private List<TrackPayload> trackEvents = new ArrayList<>();
+  private final Handler handler = new Handler();
+
+  public static Factory factory(OptimizelyManager manager) {
```

Comment:
I don't think this is ever used

Reply:
It's used by customers when register the factory with Segment.